### PR TITLE
Access docker registry anonymously

### PR DESCRIPTION
**Pull request by qiluo-msft**
### Description of PR

Summary:
We already configured the docker registry with read only access by anonymous user. So no need to `docker login` before `docker pull`.

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### CODE REVIEW

File: tests/common/system_utils/docker.py [hunk 1]
```diff
@@ -67,8 +67,8 @@ def load_docker_registry_info(dut):
             raise
 
     host = registry_vars.get("docker_registry_host")
-    username = registry_vars.get("docker_registry_username")
-    password = registry_vars.get("docker_registry_password")
+    username = None
+    password = None
```
Comment (daall) [inline tests/common/system_utils/docker.py:71]: does this work for the internal docker registry as well?

File: setup-container.sh [hunk 2]
```diff
@@ -3,8 +3,6 @@
 DOCKER_SONIC_MGMT="docker-sonic-mgmt"
 DOCKER_REGISTRY="sonicdev-microsoft.azurecr.io:443/"
 VAR_FILE="ansible/vars/docker_registry.yml"
-USER_KEY="docker_registry_username"
```
Comment (theasianpianist) [inline setup-container.sh:6]: No need to change this file, I have already merged #2068 which accomplishes anonymous pull in this script.

File: tests/common/system_utils/docker.py (hunk 1 above)
Comment (qiluo-msft) [reply tests/common/system_utils/docker.py]: revert back.

---
In reply to: [467237544](https://github.com/Azure/sonic-mgmt/pull/2069#discussion_r467237544) [](ancestors = 467237544)

File: setup-container.sh (hunk 2 above)
Comment (qiluo-msft) [reply setup-container.sh]: Rebased and reverted.

File: tests/common/system_utils/docker.py [hunk 3]
```diff
@@ -99,7 +99,6 @@ def download_image(dut, registry, image_name, image_version="latest"):
             image_version (str): The version of the image to download.
     """
 
-    dut.command("docker login {} -u {} -p {}".format(registry.host, registry.username, registry.password))
```
Comment (daall) [inline tests/common/system_utils/docker.py:102]: We still need this part if we're able to find `registry_vars` on line 54. To make this work, I think you can just check if username and password are present. If they are, then do the `login` first. If they are not, then skip the login and use the anonymous method. :)

Comment (daall) [reply tests/common/system_utils/docker.py:102]: Also you can delete the check from line 73-76 as we are now OK if username and password are not found in the hostvars.

Comment (qiluo-msft) [reply tests/common/system_utils/docker.py:102]: Deleted 2 checkers there.

---
In reply to: [467336245](https://github.com/Azure/sonic-mgmt/pull/2069#discussion_r467336245) [](ancestors = 467336245)

Comment (qiluo-msft) [reply tests/common/system_utils/docker.py:102]: Resolved

---
In reply to: [467336136](https://github.com/Azure/sonic-mgmt/pull/2069#discussion_r467336136) [](ancestors = 467336136)

**Comment by yxieca** [comment]
@theasianpianist do you have further requested change?

**Pull request merged**